MapleServer2/Types/Currency.cs

```diff
@@ -0,0 +1,70 @@
+﻿using MapleServer2.Enums;
+using MapleServer2.Packets;
+
+namespace MapleServer2.Types
+{
+    public class Currency
+    {
+        private readonly Player Player;
+        private readonly CurrencyType Type;
+        public long Amount { get; private set; }
+
+        public Currency(Player player, CurrencyType type, long input)
+        {
+            Player = player;
+            Type = type;
+            Amount = input;
+        }
+
+        public bool Modify(long input)
+        {
+            if (Amount + input < 0)
+            {
+                return false;
+            }
+            Amount += input;
+            UpdateUi();
+            return true;
+        }
+
+        public void SetAmount(long input)
+        {
+            if (input < 0)
+            {
+                return;
+            }
+            Amount = input;
+            UpdateUi();
+        }
+
+        private void UpdateUi()
```

Comment:
This function should be renamed, it's not just updating a UI value, it's sending data to the client. `UpdateClient` or similar would be more appropriate.